wee_slack.py

```diff
@@ -1445,7 +1445,8 @@ def get_typing_list(self):
 
     def mark_read(self, ts=None, update_remote=True, force=False):
         if not ts:
-            ts = SlackTS()
+            msg_timestamps = self.sorted_message_keys()
```

Comment:
How expensive is this call? Should we rather just cache this on the channel object when appending messages?

Reply:
Valid point. I have now changed `SlackChannel.messages` to be an OrderedDict and sort that when we store the messages, so we don't need to sort it on lookup.